code/game/objects/items/storage/ammo_can.dm

```diff
@@ -1,51 +1,69 @@
 //No idea why this is a toolbox but I'm not fixing that right now
-/obj/item/storage/toolbox/ammo
+/obj/item/storage/ammo_can
 	name = "ammo can"
 	desc = "A metal container for storing multiple boxes of ammunition or grenades."
 	icon_state = "ammobox"
 	item_state = "ammobox"
 	drop_sound = 'sound/items/handling/ammobox_drop.ogg'
 	pickup_sound =  'sound/items/handling/ammobox_pickup.ogg'
 	material_flags = NONE
-	has_latches = FALSE
+	drop_sound = 'sound/items/handling/toolbox_drop.ogg'
+	pickup_sound =  'sound/items/handling/toolbox_pickup.ogg'
+	w_class = WEIGHT_CLASS_BULKY
 
-/obj/item/storage/toolbox/ammo/a850r/PopulateContents()
+/obj/item/storage/ammo_can/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_volume = STORAGE_VOLUME_BACKPACK
+	STR.max_w_class = MAX_WEIGHT_CLASS_BACKPACK
+	STR.set_holdable(list(
+		/obj/item/storage/box/ammo,
+		/obj/item/mine,
+		/obj/item/grenade,
+		/obj/item/ammo_casing/caseless/rocket,
+		/obj/item/ammo_box/magazine/ammo_stack,
+		/obj/item/ammo_casing,
+		/obj/item/mine,
+		/obj/item/grenade
+		))
```

Comment:
I would go ahead and make this a var because sub types always end up needed edge cases for this.